__tests__/__unit__/job/actions.unit.test.ts

```diff
@@ -0,0 +1,475 @@
+import {
+    generateISession,
+    generateIProfile,
+    generateTreeView, generateISessionWithoutCredentials, generateTextDocument, generateInstanceOfProfile
+} from "../../../__mocks__/generators/shared";
+import { generateIJobObject, generateJobsTree } from "../../../__mocks__/generators/jobs";
+import { generateJesApi, bindJesApi } from "../../../__mocks__/generators/api";
+import * as jobActions from "../../../src/job/actions";
+import { ZoweDatasetNode } from "../../../src/dataset/ZoweDatasetNode";
+import * as dsActions from "../../../src/dataset/actions";
+import * as globals from "../../../src/globals";
+import { generateDatasetSessionNode, generateDatasetTree } from "../../../__mocks__/generators/datasets";
+import { Profiles } from "../../../src/Profiles";
+
+const activeTextEditorDocument = jest.fn();
+
+Object.defineProperty(vscode.window, "showInformationMessage", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showInputBox", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showErrorMessage", { value: jest.fn() });
+Object.defineProperty(zowe, "IssueCommand", { value: jest.fn() });
+Object.defineProperty(zowe.IssueCommand, "issueSimple", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showOpenDialog", { value: jest.fn() });
+Object.defineProperty(zowe, "GetJobs", { value: jest.fn() });
+Object.defineProperty(zowe.GetJobs, "getJclForJob", { value: jest.fn() });
+Object.defineProperty(vscode.workspace, "openTextDocument", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showTextDocument", { value: jest.fn() });
+Object.defineProperty(zowe, "ZosmfSession", { value: jest.fn() });
+Object.defineProperty(zowe.ZosmfSession, "createBasicZosmfSession", { value: jest.fn() });
+Object.defineProperty(vscode.window, "activeTextEditor", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showQuickPick", { value: jest.fn() });
+Object.defineProperty(vscode.window.activeTextEditor, "document", { get: activeTextEditorDocument });
+Object.defineProperty(globals, "LOG", { value: jest.fn() });
+Object.defineProperty(globals.LOG, "debug", { value: jest.fn() });
+Object.defineProperty(globals.LOG, "error", { value: jest.fn() });
+Object.defineProperty(Profiles, "getInstance", { value: jest.fn() });
+
+// Idea is borrowed from: https://github.com/kulshekhar/ts-jest/blob/master/src/util/testing.ts
+const mocked = (fn: any): jest.Mock => fn;
+
+describe("Job state operations", () => {
+    function generateEnvironmentalMocks() {
+        const session = generateISession();
+        const treeView = generateTreeView();
+        const iJob = generateIJobObject();
+        const imperativeProfile = generateIProfile();
+
+        return {
+            session,
+            treeView,
+            iJob,
+            imperativeProfile,
+            testJobsTree: generateJobsTree(session, iJob, imperativeProfile, treeView)
+        };
+    }
+
+    /* beforeEach(() => {
```

Comment:
What is the purpose of these comments? If the code is not used anymore, we should delete it.

Reply:
I left it to remember which mocks I need to reset. I can change it to the form of the comment.

Reply:
If you are clearing all mocks afterEach, is it still necessary to reset them beforehand?